File: src/main/java/dragonsaga/ai/DragonHumanoidAI.java
package dragonsaga.ai;

import dragonsaga.mobs.human.MartialArtistMob;
import necesse.engine.util.gameAreaSearch.GameAreaStream;
import necesse.entity.mobs.Mob;
import necesse.entity.mobs.ai.behaviourTree.composites.SelectorAINode;
import necesse.entity.mobs.ai.behaviourTree.leaves.TargetFinderAINode;
import necesse.entity.mobs.ai.behaviourTree.leaves.WandererAINode;
import necesse.entity.mobs.ai.behaviourTree.trees.PlayerChaserAI;
import necesse.entity.mobs.ai.behaviourTree.util.TargetFinderDistance;

import java.awt.*;

public class DragonHumanoidAI<T extends MartialArtistMob> extends PlayerChaserAI<T> {

    public DragonHumanoidAI(int searchDistance, int shootDistance, boolean smartPositioning, boolean changePositionOnHit, int wanderFrequency) {
        super(searchDistance, shootDistance, smartPositioning, changePositionOnHit);
        //addChild(new WandererAINode<T>(wanderFrequency));
    }

    @Override
    public GameAreaStream<Mob> streamPossibleTargets(T mob, Point base, TargetFinderDistance<T> distance) {
        return TargetFinderAINode.streamPlayersAndHumans(mob, base, distance);
    }

    @Override
    public boolean shootTarget(T t, Mob mob) {
        return false;
    }
}

File: src/main/kotlin/ai/DragonHumanoidAI.kt
package ai

import dragonsaga.mobs.MartialArtistMob
import necesse.engine.util.gameAreaSearch.GameAreaStream
import necesse.entity.mobs.Mob
import necesse.entity.mobs.ai.behaviourTree.leaves.TargetFinderAINode
import necesse.entity.mobs.ai.behaviourTree.trees.PlayerChaserAI
import necesse.entity.mobs.ai.behaviourTree.util.TargetFinderDistance
import java.awt.Point

class DragonHumanoidAI<T : MartialArtistMob?>(
    searchDistance: Int,
    shootDistance: Int,
    smartPositioning: Boolean,
    changePositionOnHit: Boolean,
    wanderFrequency: Int
) : PlayerChaserAI<T>(searchDistance, shootDistance, smartPositioning, changePositionOnHit) {
    override fun streamPossibleTargets(mob: T, base: Point, distance: TargetFinderDistance<T>): GameAreaStream<Mob> {
        return TargetFinderAINode.streamPlayersAndHumans(mob, base, distance)
    }

    override fun shootTarget(t: T, mob: Mob): Boolean {
        return false
    }
}